WISE-Community/WISE-API
# feat(Password): Increase password validation requirements

Test with
- https://github.com/WISE-Community/WISE-Client/pull/834

## Changes

- Increased server side password validation requirements to require at least 8 characters and at least one lowercase, one uppercase, and one number
- Password change endpoints now return a ResponseEntity which allows us to specify the response status code such as 200 for ok or 400 for bad request as well as send additional custom information such as what type of error occurred

## Test

- Make sure the password change endpoints enforce the new password requirements
- Make sure the password change endpoints send back an error response when an error occurs (they used to send back a success response but with error information)
- Make sure the password change success response still works

Here are the places where users can specify passwords
- student create account
- student forgot password
- teacher create account
- teacher forgot password
- teacher change student password in classroom monitor
- user edit password
- user unlink Google account

Closes #177
